fgpyo/util/inspect.py

```diff
@@ -254,9 +315,36 @@ def dict_parse(dict_string: str) -> Dict[Any, Any]:
     return parser
 
 
+def get_fields_dict(cls: Type[DataclassesOrAttrClass]) -> Dict[str, FieldType]:
+    """
+    Get the fields dict from either a dataclasses or attr dataclass.
+
+    Combine results in case someone chooses to mix them through inheritance.
+    """
+    if not (is_dataclasses_class(cls) or is_attr_class(cls)):
+        raise TypeError("cls must a dataclasses or attr class")
+    return {
+        **(get_dataclasses_fields_dict(cls) if is_dataclasses_class(cls) else {}),
+        **(get_attr_fields_dict(cls) if is_attr_class(cls) else {}),  # type: ignore
```

Comment:
Since we have such type trickery going on. Could you add a refinement to all type ignore statements in this PR? It can be done with the syntax `# type: ignore[...]` where `...` is the type checker exception kind.

Reply:
Done